_posts/2018/2018-02-18-avoid-writing-sql-when-using-activerelation.md

```diff
@@ -0,0 +1,74 @@
+---
+title: 'Avoid Writing SQL When Using ActiveRelation'
+description: 'It's fine, but prefer not to if you can.'
+layout: article
+category: ruby
+image:
+  base: '2018/avoid-writing-sql-when-using-activerelation'
+  alt: 'Coding SQL'
+  source: 'https://unsplash.com/photos/fPkvU7RDmCo'
+  credit: 'Caspar Rubin'
+---
+
+ActiveRelation, which powers the searching and querying engine of ActiveRecord, is a powerful and flexible tool.
+
+
+## Instead of...
+
+...writing literal SQL strings inside an ActiveRelation `#where` method.
+
+```ruby
+Person.where("name = #{ params[:name] } and hidden_at is null")
+```
+
+
+## Or...
+
+...writing literal strings and using the Array method to sanitise data.
+
+```ruby
+Person.where('name = ? and hidden_at is null', params[:name])
+```
+
+
+## Use...
+
+...the hash-style syntax.
+
+```ruby
+Person.where(name: params[:name], hidden_at: nil)
+```
+
+
+## Why?
+
+All methods are valid. The first two versions, where you manually write SQL, was the _only_ way before ActiveRelation was merged into Rails (in version 3.0). However, the “hash style” gives you more flexibility and safety.
```

Comment:
*All of the above methods are valid

But isn't that always the case with all of your 'instead of' things..? They're always valid, just not the best option...

I would remove this clause and start this paragraph with "The first two examples..."